File: labheranca/exer 2/Cliente.java
package lab;

import java.util.ArrayList;

public class Cliente {
    private String nome;
    private String sobrenome;
    private ArrayList<Conta> contas = new ArrayList<>();
    private int numeroDeContas;

    public Cliente(String nome, String sobrenome)
    {
        this.nome = nome;
        this.sobrenome = sobrenome;
    }
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    public String getSobrenome() {
        return sobrenome;
    }
    public void setSobrenome(String sobrenome) {
        this.sobrenome = sobrenome;
    }

    public void adicionarConta(Conta conta)
    {
        contas.add(conta);
        numeroDeContas++;
    }
    public Conta getConta(int index)
    {
        return contas.get(index);
    }
    public int getNumeroDeContas() {
        return numeroDeContas;
    }
}

File: javascript/Cliente.js
module.exports = class Cliente {
  constructor(nome, sobrenome) {
    this.nome = nome;
    this.sobrenome = sobrenome;
    this.contas = [];
  }
  getNome() {
    return this.nome;
  }
  setNome(nome) {
    this.nome = nome;
  }
  getSobrenome() {
    return this.sobrenome;
  }
  setSobrenome(sobrenome) {
    this.sobrenome = sobrenome;
  }
  getConta(indice) {
    return this.contas[indice];
  }
  getNumeroDeContas() {
    return this.contas.length;
  }
  adicionarConta(conta) {
    this.contas.push(conta);
  }
};
